lib/mix/lib/mix/scm/git.ex

```diff
@@ -232,6 +232,13 @@ defmodule Mix.SCM.Git do
     end
   end
 
+  defp redact_uri(git) do
+    case URI.parse(git) do
+      %{userinfo: nil} -> git
+      uri -> URI.to_string(%{uri | userinfo: "****:****"})
```

Comment:
Nitpick: I think the password in the `userinfo` subcomponent is optional, being `http://username@example.com`, `http://username:@example.com`, and `http://username:password@example.com` all valid. 

If that's the case, I would not use a semicolon `:` in the placeholder to avoid any confusion:

```suggestion
      uri -> URI.to_string(%{uri | userinfo: "****"})
```

Thoughts?

Reply:
@fertapric in your case there is a colon though. Plus without `:` it changes the meaning. I personally prefer the current approach. :)

Reply:
@josevalim I was mentioning that because GitHub allows to clone a repo passing a GitHub token in the _userinfo_: `git clone https://mygithubtoken@github.com/fertapric/my_repo.git` 🙂 

But I'm ok with the current placeholder 👍

Reply:
Oh, I see. The `:` is not required at all. Hrm, we could check if `:` is there or not and generate something accordingly but overall I don't think it matters? I am fine with whatever you pick. :)